x/ecocredit/genesis.go

```diff
@@ -60,6 +61,41 @@ func (s *GenesisState) Validate() error {
 		return err
 	}
 
+	// run each params validation method
+	if err := s.Params.Validate(); err != nil {
+		return err
+	}
+
+	// check that the CreditTypes in the ClassInfo slice all exist in params.CreditTypes
+	if err := validateClassInfoTypes(s.Params.CreditTypes, s.ClassInfo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateClassInfoTypes(creditTypes []*CreditType, classInfos []*ClassInfo) error {
+	typeMap := make(map[string]CreditType, len(creditTypes))
+
+	// convert to map for easier lookups
+	for _, cType := range creditTypes {
+		typeMap[cType.Abbreviation] = *cType
+	}
+
+	for _, cInfo := range classInfos {
+		// fetch param via abbreviation
+		cType, ok := typeMap[cInfo.CreditType.Abbreviation]
+
+		// if it's not found, its an invalid credit type
+		if !ok {
+			return fmt.Errorf("unknown credit type abbreviation: %s", cInfo.CreditType.Abbreviation)
```

Comment:
what about using `sdkerrors.ErrInvalidType.Wrapf("%s is not a valid credit type", creditTypeName)` instead, which is used in `serverImpl.getCreditType` to keep error msgs consistent?

Reply:
now that i think about it - perhaps ErrNotFound would be better here

Reply:
makes sense